src/1password.ts

```diff
@@ -0,0 +1,178 @@
+    for (const itemRequest of itemRequests) {
+      try {
+        const itemsJSON = await this.getVault(itemRequest.vault)
+
+        const items: Item[] = JSON.parse(itemsJSON)
+        const uuid = items
+          .filter(item => item.overview.title === itemRequest.name)
+          .map(item => item.uuid)[0]
+
+        if (!uuid) {
+          throw new Error(
+            `Could not find item in vault \u001b[38;2;255;255;0m'${itemRequest.vault}' \u001b[38;2;255;255;255mwith name \u001b[38;2;255;255;0m'${itemRequest.name}'`
+          )
+        }
+
+        core.info(`Loading ${itemRequest.name} detail`)
+        const itemJSON = await this.getItemInVault(itemRequest.vault, uuid)
+        const item: Item = JSON.parse(itemJSON)
+
+        switch (item.templateUuid) {
+          // Item
+          case '001': {
+            const username = (item.details.fields ?? []).filter(
+              field => field.designation === 'username'
+            )[0].value
+            const password = (item.details.fields ?? []).filter(
+              field => field.designation === 'password'
+            )[0].value
+
+            const usernameOutputName = `${itemRequest.outputName}_username`
+            core.setOutput(usernameOutputName, username)
+            const passwordOutputName = `${itemRequest.outputName}_password`
+            core.setSecret(password)
+            core.setOutput(passwordOutputName, password)
+
+            break
+          }
+          // Password
+          case '005': {
+            const password = item.details.password
+            if (password === undefined) {
+              throw new Error(
+                'Expected string for property item.details.password, got undefined.'
+              )
+            }
+
+            const passwordOutputName = `${itemRequest.outputName}_password`
+            core.setSecret(password)
+            core.setOutput(passwordOutputName, password)
+
+            break
+          }
+          // Document
+          case '006': {
+            const filename = item.details.documentAttributes?.fileName
+            if (filename === undefined) {
+              throw new Error(
+                'Expected string for property document.details.documentAttributes?.filename, got undefined.'
+              )
+            }
+
+            try {
+              await this.getDocument(uuid, filename)
+            } catch (error) {
+              throw new Error(
+                `Error downloading file \u001b[38;2;255;255;0m${filename} \u001b[38;2;255;255;255m\n${error}`
+              )
+            }
+
+            const documentOutputName = `${itemRequest.outputName}_filename`
+            core.setOutput(documentOutputName, filename)
+
+            break
+          }
+        }
+      } catch (error) {
+        throw new Error(error)
+      }
+    }
+  }
+
+  async getVault(vault: string): Promise<string> {
+    const env = this.onePasswordEnv
+    core.info(`Requesting 1Password Vault: ${vault}`)
```

Comment:
```suggestion
    core.info(`Getting items in 1Password Vault: ${vault}`)
```
Not a really strong opinion nor a blocker but it feels like the use of `get` and `request` could be more consistent within this file. Since the intent is to wrap `op` I'm leaning towards matching its verbs. I know it's kinda unclear because the parsed type is ItemRequest, but since that's outside the context of `op` maybe it shouldn't have an influence here.

Reply:
Yep, agree. Makes more sense in `main.ts`. 

Made it it's own function

Reply:
Yep I agree. With that - I changed `getItemsInVault` to `listItemsInVault` to more match what `op` is calling.